arithmetization/src/main/java/net/consensys/linea/zktracer/ZkTracer.java

```diff
@@ -126,7 +138,16 @@ public void traceContextExit(MessageFrame frame) {
   }
 
   /** When called, erase all tracing related to the last included transaction. */
-  public void popTransaction() {
-    hub.popTransaction();
+  public void popTransaction(final PendingTransaction pendingTransaction) {
+    if (hashOfLastTransactionTraced.equals(pendingTransaction.getTransaction().getHash())) {
```

Comment:
why do we need to check this? when would it not match?

Reply:
We have to only call the hub.pop() here if the startTransaction() for this transaction has been called. The popTransaction() is called from the plugin when the onTransactionNotSelected() in the BlockTransactionSelector is called and that can happen before the startTransaction() in the MainnetTransactionProcessor is called, e.g. because of one of the evaluateTransactionPreProcessing() calls failed.